### PR TITLE
key store & traits book

### LINKED ISSUE #749: traits & key store documentation
Describe the key store trait in the book.

### PR DESCRIPTION
**Pull request by franziskuskiefer**
Some description of the OpenMLS traits and how to use the key store.

fixes #749

### CODE REVIEW

File: openmls/tests/key_store.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        .expect("Error serializing the credential's public key.");
+
+    // Now we can store the credential_bundle.
+    backend
+        .key_store()
+        .store(id.as_slice(), &credential_bundle_to_store)
+        .expect("Failed to store credential in keystore.");
+
+    // In order to read something from the key store we need to define an ID.
+    // Here we simply take the serialized public key of the credential.
```
Comment (kkohbrok) [inline openmls/tests/key_store.rs:70]: This comment should probably go in line 57.

**Review by kkohbrok — APPROVED** [review]
More docs are always nice 🙂.

Thanks!

File: book/src/traits/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+OpenMLS defines a number of traits that have to be implemented in order to use
+OpenMLS.
+
+- [Traits](./traits.md)
+- [External Types](./types.md)
+
+## Using the key store
+
+The key store is probably one of the most interesting traits because applications
+that use OpenMLS will interact with it.
```
Comment (raphaelrobert) [inline book/src/traits/README.md:12]: Could you mention that the key is just a byte slice for all operations?

**Review by raphaelrobert — CHANGES_REQUESTED** [review]
Thanks, this looks nice! Just one nit

File: openmls/tests/key_store.rs (hunk 1 above)
Comment (franziskuskiefer) [reply openmls/tests/key_store.rs:70]: nope, 57 is not in the book (it's just setup)

**Review by raphaelrobert — APPROVED** [review]
Thanks!

**Pull request merged**